### PR TITLE
ENH better hashing for bad urls; extra test

### LINKED ISSUE #806: version number didn’t make it to pr
https://github.com/conda-forge/ccache-feedstock/pull/8

### LINKED ISSUE #817: use requests to hash urls everywhere to match conda
We still have one case of this in the updater code. IDK if this matters or not there.

cc @chrisburr who has had issues with this

### PR DESCRIPTION
**Pull request by beckermr**
I am unifying the hashing w/ a timeout and CI friendly progress meter.

closes #817 
closes #806 
related to https://github.com/regro/cf-scripts/issues/500

### CODE REVIEW

**Comment by scopatz** [comment]
Seems as though there are conflicts here

File: conda_forge_tick/hashing.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        _hash = (repr(e),)
+    finally:
+        if conn is not None:
+            conn.send(_hash)
+            conn.close()
+        else:
+            return _hash
+
+
+def hash_url(url, timeout=None, progress=False, hash_type='sha256'):
```
Comment (scopatz) [inline conda_forge_tick/hashing.py]: This seems like it could be one line in xonsh

Comment (beckermr) [reply conda_forge_tick/hashing.py]: The xonsh version is hanging the bot for more than 10 minutes on gmsh. Idk why but the above works around this with a timeout. Also we had an issue where using urllib versus requests generated different hashes. Conda build uses requests so we have to match that. 

Cc @chrisburr who tracked down the requests versus urllib issue.

Comment (beckermr) [reply conda_forge_tick/hashing.py]: See also this issue: https://github.com/regro/cf-scripts/issues/500

**Pull request merged**